File: app/src/main/java/com/lemonade/dima/repository/LemonadeRepository.java
package com.lemonade.dima.repository;

import android.support.annotation.NonNull;

import com.lemonade.dima.model.Message;
import com.lemonade.dima.model.ResponseMessages;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;

public interface LemonadeRepository {
    @NonNull
    Observable<ResponseMessages> message(Message message);

    Single<List<Message>> saved();
}

File: app/src/main/java/com/k4dima/chat/messages/repository/LemonadeRepository.kt
package com.k4dima.chat.messages.repository

import com.k4dima.chat.app.model.Message
import com.k4dima.chat.app.model.ResponseMessages

import io.reactivex.Single

interface LemonadeRepository {
    fun message(message: Message): Single<ResponseMessages>

    fun saved(): Single<MutableList<Message>>
}